src/logic/safe/store/selectors/pendingTransactions.ts

```diff
@@ -6,29 +6,43 @@ import { LocalTransactionStatus, Transaction } from 'src/logic/safe/store/models
 import { PendingTransactionsState, PENDING_TRANSACTIONS_ID } from 'src/logic/safe/store/reducer/pendingTransactions'
 import { currentChainId } from 'src/logic/config/store/selectors'
 import { ChainId } from 'src/config/chain'
+import { getTransactionByAttribute } from 'src/logic/safe/store/selectors/gatewayTransactions'
 
-export const allPendingTxs = (state: AppReduxState): PendingTransactionsState => {
+export const allPendingTxIds = (state: AppReduxState): PendingTransactionsState => {
   return state[PENDING_TRANSACTIONS_ID]
 }
 
-const pendingTxsByChain = createSelector(
-  allPendingTxs,
+export const pendingTxIdsByChain = createSelector(
+  allPendingTxIds,
   currentChainId,
   (statuses, chainId): PendingTransactionsState[ChainId] => {
     return statuses[chainId]
   },
 )
 
+export const pendingTxByChain = createSelector(
+  (state: AppReduxState) => state,
+  pendingTxIdsByChain,
+  (state: AppReduxState, pendingTxIds: PendingTransactionsState[ChainId]): Transaction | undefined => {
+    if (!pendingTxIds) {
+      return
+    }
+
+    const pendingTxId = Object.keys(pendingTxIds)[0]
```

Comment:
Have you encountered any instance when there could be more than one pending transaction?

Reply:
So far not as it should be but there might be edge cases we need to consider i.e.

1. Creating and executing two transactions in a 1/n safe in quick succession
2. Creating and executing two transactions in a 1/n safe from two different browser tabs (since we're Broadcasting the pending tx state)

Its not in the scope of this ticket but we should revisit the current data structure we use for saving pending tx(s) in the redux

Reply:
I'll mark this as resolved, but maybe it's worthwhile adding a research issue for this.